File: pascalsTriangle.java
public class Solution {
    public ArrayList<ArrayList<Integer>> generate(int numRows) {
        // Start typing your Java solution below
        // DO NOT write main() function
        ArrayList<ArrayList<Integer>> result = new ArrayList<ArrayList<Integer>>();
        if (numRows == 0) {
            return result;
        }
        ArrayList<Integer> v = new ArrayList<Integer>();
        v.add(1);
        result.add(v);
        for (int i=2; i <= numRows; i++) {
            ArrayList<Integer> v1 = new ArrayList<Integer>();
            v1.add(1);
            for (int j=1; j < i-1; j++) {
                v1.add(result.get(i-2).get(j-1) + result.get(i-2).get(j));
            }
            v1.add(1);
            result.add(v1);
        }
        return result;

    }
}

File: pascalsTriangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        vector<vector<int> > result;
        if (numRows == 0) {
            return result;
        }
        vector<int> v;
        v.push_back(1);
        result.push_back(v);
        for (int i=2; i <= numRows; i++) {
            vector<int> v;
            v.push_back(1);
            for (int j=1; j < i-1; j++) {
                v.push_back(result[i-2][j-1] + result[i-2][j]);
            }
            v.push_back(1);
            result.push_back(v);
        }
        return result;
    }
};
